### PR TITLE
Add task to generate and upload Riff-Raff artefact

### PR DESCRIPTION
**Pull request by annebyrne**
screen shot 2016-06-08 at 17 55 30

See also: https://github.com/guardian/riff-raff/pull/302

cc @rtyley @joelochlann @SiAdcock

### CODE REVIEW

File: deploy.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "defaultStacks": [ "interactives" ],
+    "packages":{
+        "interactive-brexit-companion":{
+            "type":"aws-s3",
+            "data":{
+                "prefixStage": false,
+                "prefixPackage": false,
+                "pathPrefixResource": "s3-path-prefix",
+                "bucket": "gdn-cdn",
+                "cacheControl": [
```
Comment (joelochlann) [inline deploy.json:11]: arrays of objects are generally a bit clearer indented like so:

``` javascript
"cacheControl": [
    {
        "pattern": "^\\d+/",
        "value": "max-age=315360000"
    },
    {
        "pattern": ".*",
        "value": "max-age=60"
    }
]
```

File: Gruntfile.js [hunk 2]
```diff
@@ -103,14 +115,18 @@ module.exports = function(grunt) {
...
                     },
                     { // ASSETS
                         expand: true, cwd: 'build/',
                         src: ['main.js', 'main.css', 'main.js.map', 'main.css.map',
                             'embed.js', 'embed.css', 'embed.js.map', 'embed.css.map',
                             'assets/**/*'],
-                        dest: 'deploy/<%= visuals.timestamp %>/<%= visuals.timestamp %>'
+                        dest: 'deploy/packages/interactive-brexit-companion/<%= visuals.timestamp %>'
+                    },
+                    {   expand: true, cwd: '.',
```
Comment (joelochlann) [inline Gruntfile.js]: generally you'd start the object properties on the following line, unless you were putting the whole object on one line, e.g.

``` javascript
var foo = [
    {a: 10, b: 20},
    {a: 11, b: 21},
    {a: 12, b: 22}
];
```

or

``` javascript
var foo = [
    {
        a: 10,
        b: 20
    },
    {
        a: 11,
        b: 21
    },
    {
        a: 12,
        b: 22
    }
];
```

**Comment by joelochlann** [comment]
Looks good to me, besides a few pedantic whitespace style comments. However I don't really know enough about RiffRaff or TeamCity to make any substantive comment on this one. I'd definitely want @rtyley and/or @philwills to take a look before merging

File: package.json [hunk 3]
```diff
@@ -11,9 +11,11 @@
     "eslint-import-resolver-jspm": "^2.1.0",
     "eslint-plugin-import": "^1.8.0",
     "grunt": "^0.4.5",
+    "grunt-aws": "^0.5.4",
```
Comment (philwills) [inline package.json]: Is this actually being used?

**Comment by philwills** [comment]
Looks good to me.

**Comment by annebyrne** [comment]
Tidied code based on above comments & merged 👍

**Pull request merged**